lib/phlex/kit.rb

```diff
@@ -1,31 +1,39 @@
 # frozen_string_literal: true
 
 module Phlex::Kit
-	def self.extended(mod)
-		warn "⚠️ [WARNING] Phlex::Kit is experimental and may be removed from future versions of Phlex."
-		super
+	module LazyLoader
+		def method_missing(name, *args, **kwargs, &block)
+			if name[0] == name[0].upcase && __phlex_kit_constants__.include?(name) && __get_phlex_kit_constant__(name) && methods.include?(name)
```

Comment:
I spent way too long trying to performance golf `name[0] == name[0].upcase`.

I thought it was going to be easier than it was when I remembered that Ruby supports non-ascii capital letters as well. I did manage to figure out how to regex it, which does perform about twice as fast as this string transformation + comparison. But after all that, I think what you have is better, because it's clearer what is being done.

```rb
name.match?(/\A(\p{Lu}|\p{Lt})/) # \p{} checks for unicode properties, Lu is uppercase, Lt is titlecase

# Comparison:
#                regex: 13742991.5 i/s
#          start_with?:  7782039.9 i/s - 1.77x  slower
#    [0] == [0].upcase:  6729981.7 i/s - 2.04x  slower
```

Reply:
Isn't `name[0].casecmp?(name[0])` the fastest way to do a case-insensitive comparison?

Reply:
Oh, I misunderstood what this was doing. Nevermind.